extensions/amp-consent/amp-consent.md

```diff
@@ -98,35 +98,49 @@ Currently, the `consents` object only supports a single consent instance. A cons
 
 #### checkConsentHref
 
-`checkConsentHref`: Instructs AMP to make a CORS POST request with credentials to the specified URL to remotely configure the consent. The purpose is to determine if a prompt UI should be shown if the consent state is unknown.
-
-`checkConsentHref` is required if [`promptIfUnknownForGeoGroup`](#promptifunknownforgeogroup) is not defined.
+`checkConsentHref`: Instructs AMP to make a CORS POST request with credentials to the specified URL to remotely configure the consent. The purpose is to 1) determine if consent is required for the user, 2) get the consent state from server, 3) get extra `sharedData` from server to share with AMP components.
 
 ##### Request
 
 AMP sends the consent instance ID in the `consentInstanceId` field with the POST request.
 
 ```html
-{ "consentInstanceId": "my-consent" }
+{ 
+  "consentInstanceId": {string},
+  "consentStateValue": {enum}, // the stored consent state in client cache
+                               // takes value of ["accepted", "rejected", "unknown"]
+  "consentString": {string},   // the stored consent string in client cache
+  "matchedGeoGroup": {string}, // (new) the user's geoGroup detected by AMP.
+}
 ```
 
 ##### Response
 
-AMP expects the response to be a JSON object with a `promptIfUnknown` value that tells AMP if a prompt should be displayed when the user consent state is unknown.
+AMP expects the response to be a JSON object like the following:
 
 ```html
-{ "promptIfUnknown": true/false }
+{ 
+  "consentRequired": {boolean}              // whether consent is required from the user. Previously named `promptIfUnknown`
+  "consentStateValue": {enum} [default: null], // (new) the latest consent state known by the server
+                                            // takes value of ["accepted", "rejected", "unknown"]
+                                            // The value will be automatically used to update the client cache
+                                            // if it's one of ["accepted", "rejected"]
+  "consentString": {string},                // (new) the latest consent string known by the server
+  "expireCache": {boolean} [default: false] // (new) indicate that the cache needs to be cleared.  
+                                            // Set to `true` to enforce server side consent state. 
```

Comment:
Maybe for clarification: 
Set to `true` in conjunction with consentStateValue='accepted'/'rejected', to enforce server side consent state